app/controllers/guides_controller.rb

```diff
@@ -6,196 +6,98 @@ def index
 
   def new
     type = params[:type].presence_in(%w{ GuideCommunity Point })
+    guide = Guide.new(type: type)
 
     @guide_form = GuideForm.new(
-      guide: Guide.new(type: type),
+      guide: guide,
       edition: Edition.new,
       user: current_user,
       )
   end
 
   def create
     guide = Guide.new(type: guide_form_params[:type])
-    edition = guide.editions.build
-    @guide_form = GuideForm.new(
-      guide: guide,
-      edition: edition,
-      user: current_user,
-      )
-    @guide_form.assign_attributes(guide_form_params)
 
-    publication = Publisher.new(content_model: @guide_form)
-                    .save_draft(GuideFormPublicationPresenter.new(@guide_form))
-    if publication.success?
-      redirect_to edit_guide_path(@guide_form), notice: 'Guide has been created'
-    else
-      flash.now[:error] = publication.error
-      render 'new'
-    end
+    save(guide)
   end
 
   def edit
     guide = Guide.find(params[:id])
-    edition = guide.latest_edition
 
     @guide_form = GuideForm.new(
       guide: guide,
-      edition: edition,
+      edition: guide.latest_edition,
       user: current_user
       )
   end
 
   def update
     guide = Guide.find(params[:id])
-    edition = guide.editions.build(guide.latest_edition.dup.attributes)
-    edition.created_by = current_user
-
-    @guide_form = GuideForm.new(
-      guide: guide,
-      edition: edition,
-      user: current_user
-      )
 
     if params[:send_for_review].present?
-      send_for_review(guide, edition)
+      manage!(guide, :request_review, message: "A review has been requested")
     elsif params[:approve_for_publication].present?
-      approve_for_publication(guide, edition)
+      manage!(guide, :approve_for_publication, message: "Thanks for approving this guide")
     elsif params[:publish].present?
-      publish
+      manage(guide, :publish, message: "Guide has been published")
     elsif params[:discard].present?
-      discard
+      manage(guide, :discard_draft, message: "Draft has been discarded", redirect: root_path)
     else
-      save_draft
+      save(guide)
     end
   end
 
 private
 
-  def send_for_review(guide, edition)
-    edition.state = 'review_requested'
-    edition.created_by = current_user
-    edition.save!
+  def manage!(guide, action, opts = {})
```

Comment:
I wanted to do that to :) I couldn't find a more readable way and got bored because it was only a few lines. Got a suggestion for me? (Don't waste your time on it though..)